script/tool/lib/src/common/package_command.dart

```diff
@@ -436,16 +436,26 @@ abstract class PackageCommand extends Command<void> {
       packages = <String>{currentPackageName};
     }
 
-    Set<String> excludedPackageNames = getExcludedPackageNames();
-
-    final Set<String> filter =
-        _expandYamlInPackageList(getStringListArg(_filterPackagesArg));
-    if (filter.isNotEmpty) {
-      final List<String> sortedList = filter.toList()..sort();
+    final Set<String> excludedPackageNames = getExcludedPackageNames();
+    final bool hasFilter = argResults?.wasParsed(_filterPackagesArg) ?? false;
+    final Set<String>? excludeAllButPackageNames = hasFilter
+        ? _expandYamlInPackageList(getStringListArg(_filterPackagesArg))
+        : null;
+    if (excludeAllButPackageNames != null &&
+        excludeAllButPackageNames.isNotEmpty) {
+      final List<String> sortedList = excludeAllButPackageNames.toList()
+        ..sort();
       print('--$_filterPackagesArg is excluding packages that are not '
           'included in: ${sortedList.join(',')}');
-      excludedPackageNames =
-          excludedPackageNames.union(packages.difference(filter));
+    }
+    // Utility to return true if a package that could be identified by any of
```

Comment:
This is funky, but not a deal breaker imo

Reply:
I don't tend to use local functions very often, but they can be nice for small local code sharing.

Reply:
I meant the comment itself, sorry. I thought my highlight would come through. 

> Utility to return true

Reply:
Oh, I see. Agreed; changed to just "Returns true if [...]".